test/tools/TestExe/TestExe.cs

```diff
@@ -40,6 +47,48 @@ static void EchoArgs(string[] args)
             }
         }
 
+        static void WriteStandardOutput()
+        {
+            // write a string of characters (bytes), this closest resembles a binary
+            // write the string "test" with an accent over the e
+            byte[] testbytes = new byte[] { 116, 233, 115, 116 };
+            foreach(byte b in testbytes)
+            {
+                Console.Write((char)b);
+            }
+        }
+
+        static void ReadStandardInput(string[] args)
+        {
+            Encoding encodingToUse = null;
+            if (args.Length > 1)
+            {
+                // the encoding page to get could be provided as an argument;
+                int result;
+                if ( int.TryParse(args[1], out result))
+                {
+                    try
+                    {
+                        encodingToUse = Encoding.GetEncoding(result);
+                    }
+                    catch
+                    {
+                        ;
```

Comment:
Seems like you should fail this and return non-zero exit code if invalid encoding is provided.  Also suggest using the string overload version instead of the int making the code easier to read.

Reply:
i've replaced this with something far simpler (see dongbo's suggestion above)